### PR TITLE
Bug/26814 hpa  restart count is also counting the familiarisation check

### PR DESCRIPTION
**Pull request by jon-shipley**
Restart fixes:

* allow a restart to be removed up until the associated check is started
* use URL slugs instead of DB Ids when deleting a restart
* allow the correct number of restarts as well as the *original* check
* add `jq` as a dependency when running `create-items.sh`

### CODE REVIEW

File: admin/services/restart.service.js [hunk 1]
```diff
@@ -191,14 +195,26 @@ restartService.getStatus = async pupilId => {
...
  * @returns {String}
  */
+restartService.markDeleted = async (pupilUrlSlug, userId, schoolId) => {
+  const pupil = await pupilDataService.sqlFindOneBySlug(pupilUrlSlug, schoolId)
+
+  if (featureToggles.isFeatureEnabled('prepareCheckMessaging')) {
+    // see if there is a check associated with this restart, ideally the slug
+    // would refer to the restart itself, and not the pupil.
+    const restart = await pupilRestartDataService.sqlFindOpenRestartForPupil(pupilUrlSlug, schoolId)
+    if (restart.check_id) {
```
[the remark this answers is not in the document]
Comment (jon-shipley) [reply admin/services/restart.service.js]: Yes, once the HPA is mature we will remove the legacy architecture, and the feature toggle 'if' can be removed.

File: functions/check-started/index.js [hunk 2]
```diff
@@ -114,3 +107,27 @@ async function updateAdminDatabaseForCheckStarted (checkCode, startedAt, logger)
...
+  return code === 'COL'
+}
+
+/**
+ * Place a request for a pupil status change on the queue
+ * @param pupilId
+ * @param checkCode
+ * @return {Promise<*>}
+ */
+async function requestPupilStatusChange (pupilId, checkCode) {
```
[the remark this answers is not in the document]
Comment (jon-shipley) [reply functions/check-started/index.js]: Because it reads better at the higher level?

**Comment by GuyHarwood** [comment]
@jon-shipley would you be able to add json file reading to `create-items.sh` as part of this?

**Comment by jon-shipley** [comment]
@GeorgeChatzigiannis - Good spot!  `create-items.sh` now pulls the queues and tables in from the json file, so the json can be authoritative.

**Pull request merged**